### ISSUE TITLE
Failed to minify the code from /lib/base.js

### ISSUE THREAD

**Issue by darioielardi** [feathersjs/rest-client#14]
### Steps to reproduce

Generate a create-react-app with @feathers/client with rest configured and "npm run build".

### Expected behavior

A build pack is generated.

### Actual behavior
"Failed to compile. Failed to minify the code from this file: 
./node_modules/@feathersjs/rest-client/lib/base.js:2"

##
Thank you for developing feathersjs, you're awesome.

**Comment by daffl** [comment]
Since the new packages are not pre-transpiled, the Webpack Babel build configuration has to include `node_modules/@feathersjs` like this:

```js
  module: {
    rules: [{
      test: /\.jsx?$/,
      exclude: /node_modules\/(?!(@feathersjs/))/,
      loader: 'babel-loader'
    }]
  },
```

If you don't want to deal with the chaos of an ejected `create-react-app` you probably want to use https://github.com/feathersjs/client (`@feathersjs/dist/feathers`)

**Comment by darioielardi** [comment]
Thank you @daffl, with 'feathers-client', 'feathers-hooks' and 'feathers-authentication-client' I was able to solve the problem and get a production build.
But now the problem is that in the app ( deployed to Heroku ) I can't auth a user, cause seems like feathers is not able to store jwt:
![capture](https://user-images.githubusercontent.com/19256200/32813853-8e6ee612-c9ac-11e7-9cba-42a85bac726e.PNG)
In chrome devtools I can't see 'feathers-jwt' in cookies or localstorage, and I still have my previous tested code (tested only in development, not on Heroku).
Is it possible that changing back to old packages causes this problem?

My feathers client code:
```
import io from 'socket.io-client'
import feathers from 'feathers-client'
import hooks from 'feathers-hooks'
import socketio from 'feathers-socketio/client'
import auth from 'feathers-authentication-client'

const apiURL = process.env.NODE_ENV === 'production' ? 'https://my-server.herokuapp.com' : 'http://localhost:5000'

const socket = io(apiURL, {
  transports: ['websocket']
})

let client = feathers()
  .configure(hooks())
  .configure(socketio(socket))
  .configure(auth({
    storage: window.localStorage,
    cookie: 'feathers-jwt'
  }))

export default client
```

**Comment by daffl** [comment]
Odd, it should be backwards compatible but to verify (I didn't get to updating the client documentation yet) try the latest version like this:

```js
import io from 'socket.io-client'
import feathers from '@feathersjs/client/dist/feathers'

const { socketio, authentication } = feathers
const apiURL = process.env.NODE_ENV === 'production' ? 'https://my-server.herokuapp.com' : 'http://localhost:5000'

const socket = io(apiURL, {
  transports: ['websocket']
})

let client = feathers()
  .configure(socketio(socket))
  .configure(authentication({
    storage: window.localStorage,
    cookie: 'feathers-jwt'
  }))

export default client
```

If you are still getting the same error it might be something else.

**Comment by darioielardi** [comment]
Build packing still works with '@feathersjs/client/dist/feathers', but the cookies problem still occurs. Any suggestions?

**UPDATE**
I just realized that with the above code ( using '@feathersjs/client/dist/feathers' ) I'm not able to use feathers services. 
`Timeout of 5000ms exceeded calling find on *service*`

So I changed back again to 'feathers-client' and everything works. Auth on Heroku still not works. Any suggestions?

**Comment by daffl** [comment]
Please open a new issue with steps to reproduce if it still persists. I added docs for module loader integration so we can close this one.

**Comment by eric-burel** [comment]
Hi @daffl , could you please reopen this issue? I provided you a repo for reproduction [https://github.com/lebrun-burel/demo-feathers-fail](https://github.com/lebrun-burel/demo-feathers-fail). You just need to install and run `npm run build`. The exact same issue still happens, and of course I'd like to avoid to eject if its possible. Do you have an idea on how to fix this?

**Comment by eric-burel** [comment]
I quickfixed this by forking the lib and transpiling them but this is not optimal :/ Is there a specific reason for not providing a ES5 version? I am not fond of requiring user to use webpack/babel to get ES5 js, see the issues it can raise with 3rd party tools like create-react-app.

**Comment by daffl** [comment]
All Feathers libraries are using CommonJS so I don't think there is a way around using Webpack/Babel anyway. As mentioned in the [create-react-app section](https://docs.feathersjs.com/api/client.html#create-react-app) the ES5 builds are provided in [@feathersjs/client](https://docs.feathersjs.com/api/client.html#feathersjsclient). A specific build for the REST client can be found in `@feathersjs/client/rest`.

**Comment by eric-burel** [comment]
I totally missed the fact that you could use this build as a npm package, I was reading this repo README that only talks about either the ES6 lib or the CDN and could not found it. Thanks for pointing out the link, I'll confirm if this work in a few instants.

Edit: I updated the doc to make the create-react-app import process explicit. It works fine, thanks :)

**Comment by brianjohnhanna** [comment]
The `@feathersjs/client` does indeed contain all the ES5 compiled packages, however I don't see an ES5 version of the `@feathersjs/errors` in there and therefore am having same build errors with `create-react-app` while referencing the errors lib.

I solved it and hopefully any future ES6/ES5 issues by following the instructions here: https://daveceddia.com/customize-create-react-app-webpack-without-ejecting/

1. I used `npm install --save-dev babel-core babel-loader babel-preset-env` to install the dependencies.
2. Added `.babelrc` with only the line `{ "presets": ["env"] }`.
3. Added the `config-overrides.prod.js` (described in article) which looks like this:

```
module.exports = function(config) {
  // Add the babel loader second-to-last
  // (last one must remain as the "file-loader")
  let loaderList = config.module.rules[1].oneOf;
  loaderList.splice(loaderList.length - 1, 0, {
    test: /\.jsx?$/,
    exclude: /node_modules\/(?!(@feathersjs)\/).*/,
    loader: 'babel-loader'
  });
}
```
4. Updated the `build` script in `package.json` to `node scripts/customized-config build` as described in the article.

Everything works fine now, the build throws no errors.
